// fitbit-api-client/dist/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CLASSIC_SLEEP_LEVELS = exports.STAGES_SLEEP_LEVELS = exports.FITBIT_SCOPES = exports.WEIGHT_SCOPE = exports.TEMPERATURE_SCOPE = exports.SOCIAL_SCOPE = exports.SLEEP_SCOPE = exports.SETTINGS_SCOPE = exports.RESPIRATORY_RATE_SCOPE = exports.PROFILE_SCOPE = exports.OXYGEN_SATURATION_SCOPE = exports.NUTRITION_SCOPE = exports.LOCATION_SCOPE = exports.HEARTRATE_SCOPE = exports.ELECTROCARDIOGRAM_SCOPE = exports.CARDIO_FITNESS_SCOPE = exports.ACTIVITY_SCOPE = exports.CODE_CHALLENGE_METHOD = exports.FITBIT_AUTH_URL = exports.FITBIT_API_BASE_URL = void 0;
// API base URL
exports.FITBIT_API_BASE_URL = 'https://api.fitbit.com';
// 認可ページURL
exports.FITBIT_AUTH_URL = 'https://www.fitbit.com/oauth2/authorize';
// OAuthで使用する定数
exports.CODE_CHALLENGE_METHOD = 'S256';
exports.ACTIVITY_SCOPE = 'activity';
exports.CARDIO_FITNESS_SCOPE = 'cardio_fitness';
exports.ELECTROCARDIOGRAM_SCOPE = 'electrocardiogram';
exports.HEARTRATE_SCOPE = 'heartrate';
exports.LOCATION_SCOPE = 'location';
exports.NUTRITION_SCOPE = 'nutrition';
exports.OXYGEN_SATURATION_SCOPE = 'oxygen_saturation';
exports.PROFILE_SCOPE = 'profile';
exports.RESPIRATORY_RATE_SCOPE = 'respiratory_rate';
exports.SETTINGS_SCOPE = 'settings';
exports.SLEEP_SCOPE = 'sleep';
exports.SOCIAL_SCOPE = 'social';
exports.TEMPERATURE_SCOPE = 'temperature';
exports.WEIGHT_SCOPE = 'weight';
// スコープ一覧
exports.FITBIT_SCOPES = [
    exports.ACTIVITY_SCOPE,
    exports.CARDIO_FITNESS_SCOPE,
    exports.ELECTROCARDIOGRAM_SCOPE,
    exports.HEARTRATE_SCOPE,
    exports.LOCATION_SCOPE,
    exports.NUTRITION_SCOPE,
    exports.OXYGEN_SATURATION_SCOPE,
    exports.PROFILE_SCOPE,
    exports.RESPIRATORY_RATE_SCOPE,
    exports.SETTINGS_SCOPE,
    exports.SLEEP_SCOPE,
    exports.SOCIAL_SCOPE,
    exports.TEMPERATURE_SCOPE,
    exports.WEIGHT_SCOPE,
];
exports.STAGES_SLEEP_LEVELS = ['deep', 'light', 'rem', 'wake'];
exports.CLASSIC_SLEEP_LEVELS = ['restless', 'asleep', 'awake'];
