// fitbit-api-client/dist/models/hrv-intraday.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HRVIntradayResponseFromJson = HRVIntradayResponseFromJson;
const types_utils_1 = require("../utils/types.utils");
const date_utils_1 = require("../utils/date.utils");
function HRVIntradayResponseFromJson(offsetFromUTCMillis, json) {
    const hrvData = (0, types_utils_1.exists)(json, 'hrv') ? (0, types_utils_1.get)(json, 'hrv') : [];
    const hRVIntraday = hrvData.map((data) => HRVIntradayFromJson(data, offsetFromUTCMillis));
    return {
        hRVIntraday: hRVIntraday,
    };
}
function HRVIntradayFromJson(json, offsetFromUTCMillis) {
    return {
        localDate: (0, types_utils_1.get)(json, 'dateTime'),
        minutes: (0, types_utils_1.get)(json, 'minutes').map((data) => HRVIntradayMinuteDataFromJson(data, offsetFromUTCMillis)),
    };
}
function HRVIntradayMinuteDataFromJson(json, offsetFromUTCMillis) {
    return {
        minute: (0, date_utils_1.convertToOffsetDate)(new Date(`${(0, types_utils_1.get)(json, 'minute')}Z`), offsetFromUTCMillis),
        value: HRVIntradayValueDataFromJson((0, types_utils_1.get)(json, 'value')),
    };
}
function HRVIntradayValueDataFromJson(json) {
    return {
        rmssd: (0, types_utils_1.get)(json, 'rmssd'),
        coverage: (0, types_utils_1.get)(json, 'coverage'),
        hf: (0, types_utils_1.get)(json, 'hf'),
        lf: (0, types_utils_1.get)(json, 'lf'),
    };
}
