// fitbit-api-client/dist/models/activities/calories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaloriesResponseFromJson = CaloriesResponseFromJson;
const types_utils_1 = require("../../utils/types.utils");
const date_utils_1 = require("../../utils/date.utils");
function CaloriesResponseFromJson(localDate, offsetFromUTCMillis, json) {
    const activitiesCalories = (0, types_utils_1.get)(json, 'activities-calories').map((data) => CaloriesDailyDataFromJson(data));
    const activitiesCaloriesIntraday = (0, types_utils_1.exists)(json, 'activities-calories-intraday')
        ? ActivitiesCaloriesIntradayFromJson(localDate, offsetFromUTCMillis, (0, types_utils_1.get)(json, 'activities-calories-intraday'))
        : undefined;
    return {
        activitiesCalories,
        activitiesCaloriesIntraday,
    };
}
function CaloriesDailyDataFromJson(json) {
    return {
        calories: Number((0, types_utils_1.get)(json, 'value')),
        localDate: (0, types_utils_1.get)(json, 'dateTime'),
    };
}
function ActivitiesCaloriesIntradayFromJson(localDate, offsetFromUTCMillis, json) {
    return {
        dataset: (0, types_utils_1.get)(json, 'dataset').map((data) => ActivitiesCaloriesIntradayDataFromJson(localDate, offsetFromUTCMillis, data)),
        datasetInterval: (0, types_utils_1.get)(json, 'datasetInterval'),
        datasetType: (0, types_utils_1.get)(json, 'datasetType'),
    };
}
function ActivitiesCaloriesIntradayDataFromJson(localDate, offsetFromUTCMillis, json) {
    return {
        dateTime: (0, date_utils_1.convertToOffsetDate)(new Date(`${localDate}T${(0, types_utils_1.get)(json, 'time')}Z`), offsetFromUTCMillis),
        calories: (0, types_utils_1.get)(json, 'value'),
    };
}
