// fitbit-api-client/dist/models/activities/steps.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StepsResponseFromJson = StepsResponseFromJson;
const types_utils_1 = require("../../utils/types.utils");
const date_utils_1 = require("../../utils/date.utils");
function StepsResponseFromJson(localDate, offsetFromUTCMillis, json) {
    const activitiesSteps = (0, types_utils_1.get)(json, 'activities-steps').map((data) => StepsDailyDataFromJson(data));
    const activitiesStepsIntraday = (0, types_utils_1.exists)(json, 'activities-steps-intraday')
        ? ActivitiesStepsIntradayFromJson(localDate, offsetFromUTCMillis, (0, types_utils_1.get)(json, 'activities-steps-intraday'))
        : undefined;
    return {
        activitiesSteps,
        activitiesStepsIntraday,
    };
}
function StepsDailyDataFromJson(json) {
    return {
        steps: Number((0, types_utils_1.get)(json, 'value')),
        localDate: (0, types_utils_1.get)(json, 'dateTime'),
    };
}
function ActivitiesStepsIntradayFromJson(localDate, offsetFromUTCMillis, json) {
    return {
        dataset: (0, types_utils_1.get)(json, 'dataset').map((data) => StepsIntradayDataFromJson(localDate, offsetFromUTCMillis, data)),
        datasetInterval: (0, types_utils_1.get)(json, 'datasetInterval'),
        datasetType: (0, types_utils_1.get)(json, 'datasetType'),
    };
}
function StepsIntradayDataFromJson(localDate, offsetFromUTCMillis, json) {
    return {
        dateTime: (0, date_utils_1.convertToOffsetDate)(new Date(`${localDate}T${(0, types_utils_1.get)(json, 'time')}Z`), offsetFromUTCMillis),
        steps: (0, types_utils_1.get)(json, 'value'),
    };
}
