// fitbit-api-client/dist/apis/activity.api.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActivityApi = void 0;
const base_api_1 = require("./base.api");
const models_1 = require("../models");
const models_2 = require("../models");
const date_utils_1 = require("../utils/date.utils");
class ActivityApi extends base_api_1.BaseApi {
    constructor() {
        super(...arguments);
        this.scope = 'activity';
    }
    /**
     * 歩数記録取得API
     * @param request
     * @param offsetFromUTCMillis
     * @param options
     */
    async getStepsIntradayByDate(request, offsetFromUTCMillis, options) {
        return (0, models_1.StepsResponseFromJson)(request.localDate, offsetFromUTCMillis, await this.getStepsIntradayByDateRaw(request, options));
    }
    async getStepsIntradayByDateRaw(request, options) {
        return this.getActivityIntradayByDate({ ...request, resource: 'steps' }, options);
    }
    /**
     * カロリー記録取得API
     * @param request
     * @param offsetFromUTCMillis
     * @param options
     */
    async getCaloriesIntradayByDate(request, offsetFromUTCMillis, options) {
        return (0, models_2.CaloriesResponseFromJson)(request.localDate, offsetFromUTCMillis, await this.getCaloriesIntradayByDateRaw(request, options));
    }
    async getCaloriesIntradayByDateRaw(request, options) {
        return this.getActivityIntradayByDate({ ...request, resource: 'calories' }, options);
    }
    async getDistanceIntradayByDateRaw(request, options) {
        return this.getActivityIntradayByDate({ ...request, resource: 'distance' }, options);
    }
    async getElevationIntradayByDateRaw(request, options) {
        return this.getActivityIntradayByDate({ ...request, resource: 'elevation' }, options);
    }
    async getFloorsIntradayByDateRaw(request, options) {
        return this.getActivityIntradayByDate({ ...request, resource: 'floors' }, options);
    }
    async getSwimmingStrokesByDateRaw(request, options) {
        return this.getActivityIntradayByDate({ ...request, resource: 'swimming-strokes' }, options);
    }
    /**
     * ActivityAPI
     * https://dev.fitbit.com/build/reference/web-api/activity/
     * @param request
     * @param options
     */
    async getActivityIntradayByDate(request, options) {
        const { resource, localDate, detailLevel } = request;
        (0, date_utils_1.validateDateString)(localDate);
        const path = `/1/user/-/activities/${resource}/date/${localDate}/1d/${detailLevel}.json`;
        return this.get(path, options);
    }
}
exports.ActivityApi = ActivityApi;
