// fitbit-api-client/dist/apis/heart-rate.api.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HeartRateApi = void 0;
const base_api_1 = require("./base.api");
const models_1 = require("../models");
const date_utils_1 = require("../utils/date.utils");
class HeartRateApi extends base_api_1.BaseApi {
    constructor() {
        super(...arguments);
        this.scope = 'heartrate';
    }
    /**
     * 心拍数取得API
     * https://dev.fitbit.com/build/reference/web-api/intraday/get-heartrate-intraday-by-date/
     * @param request
     * @param offsetFromUTCMillis
     * @param options
     */
    async getHeartRateIntradayByDate(request, offsetFromUTCMillis, options) {
        const { localDate } = request;
        return (0, models_1.HeartRateResponseFromJson)(localDate, offsetFromUTCMillis, await this.getHeartRateIntradayByDateRaw(request, options));
    }
    async getHeartRateIntradayByDateRaw(request, options) {
        const { localDate, detailLevel } = request;
        (0, date_utils_1.validateDateString)(localDate);
        const path = `/1/user/-/activities/heart/date/${localDate}/1d/${detailLevel}.json`;
        return this.get(path, options);
    }
    /**
     * https://dev.fitbit.com/build/reference/web-api/heartrate-variability/get-hrv-summary-by-date/
     * @param request
     * @param options
     */
    async getHRVSummaryByDate(request, options) {
        return (0, models_1.HRVSummaryResponseFromJson)(await this.getHRVSummaryByDateRaw(request, options));
    }
    async getHRVSummaryByDateRaw(request, options) {
        const { localDate } = request;
        (0, date_utils_1.validateDateString)(localDate);
        const path = `/1/user/-/hrv/date/${localDate}.json`;
        return this.get(path, options);
    }
    /**
     * https://dev.fitbit.com/build/reference/web-api/intraday/get-hrv-intraday-by-date/
     * @param request
     * @param offsetFromUTCMillis
     * @param options
     */
    async getHRVIntradayByDate(request, offsetFromUTCMillis, options) {
        return (0, models_1.HRVIntradayResponseFromJson)(offsetFromUTCMillis, await this.getHRVIntradayByDateRaw(request, options));
    }
    async getHRVIntradayByDateRaw(request, options) {
        const { localDate } = request;
        (0, date_utils_1.validateDateString)(localDate);
        const path = `/1/user/-/hrv/date/${localDate}/all.json`;
        return this.get(path, options);
    }
}
exports.HeartRateApi = HeartRateApi;
