// fitbit-api-client/dist/apis/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.bodyApi = exports.spO2Api = exports.temperatureApi = exports.sleepApi = exports.activityApi = exports.heartRateApi = exports.profileAPi = exports.oauthApi = void 0;
const oauth_api_1 = require("./oauth.api");
const heart_rate_api_1 = require("./heart-rate.api");
const activity_api_1 = require("./activity.api");
const sleep_api_1 = require("./sleep.api");
const temperature_api_1 = require("./temperature.api");
const spo2_api_1 = require("./spo2.api");
const profile_api_1 = require("./profile.api");
const body_api_1 = require("./body.api");
exports.oauthApi = new oauth_api_1.OAuthApi();
exports.profileAPi = new profile_api_1.ProfileApi();
exports.heartRateApi = new heart_rate_api_1.HeartRateApi();
exports.activityApi = new activity_api_1.ActivityApi();
exports.sleepApi = new sleep_api_1.SleepApi();
exports.temperatureApi = new temperature_api_1.TemperatureApi();
exports.spO2Api = new spo2_api_1.SpO2Api();
exports.bodyApi = new body_api_1.BodyApi();
